### PR TITLE
Atomic writing of files on Windows with a specific security descriptor

### PR DESCRIPTION
**Pull request by amartinezfayo**
Current `AtomicWriteFile` function in the `diskutil` package does not allow to define a specific security descriptor to be used in the files written on Windows.
In SPIRE, we call this function to either atomically write files with permissions granted to read/write to the owner only, or with permissions granted to read/write to the owner and to read by everyone.
This PR makes the changes to have two separate functions that have os-specific implementations: `AtomicWritePrivateFile` and `AtomicWritePubliclyReadableFile` that respect the intended permissions in each platform.

Fixes part of #3189.

### CODE REVIEW

**Review by azdagron — COMMENTED** [review]
Thanks, @amartinezfayo! The new function names are very clear in intent and increase the readability, IMO. Nice work. I have only one material comment around the closing of a handle in the unit tests. The rest are nitpicks :)

File: pkg/common/diskutil/file_windows_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+				windows.GENERIC_WRITE,
+				0,
+				nil,
+				windows.OPEN_EXISTING,
+				windows.FILE_ATTRIBUTE_NORMAL,
+				0)
+
+			require.NoError(t, err)
+			sd, err := windows.GetSecurityInfo(handle, windows.SE_FILE_OBJECT, windows.DACL_SECURITY_INFORMATION)
+			require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (amartinezfayo) [reply pkg/common/diskutil/file_windows_test.go:83]: >If it needs to be closed before the rest of the test runs then we can still close on line 84

This is indeed the case. I think you meant closing on this line 83 before this assertion? That would be simple and would work well.

File: pkg/common/diskutil/file_windows.go [hunk 2]
```diff
@@ -17,12 +18,31 @@ const (
 	movefileWriteThrough    = 0x8
 )
 
-// AtomicWriteFile writes data out.  It writes to a temp file first, fsyncs that file,
-// then swaps the file in. Rename file using a custom MoveFileEx that uses 'MOVEFILE_WRITE_THROUGH' witch waits until
-// file is synced to disk.
-func AtomicWriteFile(path string, data []byte, mode os.FileMode) error {
+type fileWithSecurityAttr struct {
```
[the remark this answers is not in the document]
Comment (amartinezfayo) [reply pkg/common/diskutil/file_windows.go:21]: Good call. Naming it `fileAttribs` would be more clear.

**Review by azdagron — APPROVED** [review]
\o/

**Pull request merged**